src/business/editing.js

```diff
@@ -0,0 +1,44 @@
+import {addIndex, assoc, curry, evolve, inc, has, map, omit, pipe, when} from "ramda";
+
+
+const mapWithIndex = addIndex(map);
+
+/*
+Utilities for identifying meals and removing IDs before saving,
+cause that's just how i'm doing it for now.
+*/
+const identifyMeal = (meal, id) => Object.assign({id}, meal);
+
+const identifyMealsList = mapWithIndex(identifyMeal);
+
+const unIdentifyMeal = omit(["id"]);
+
+const unIdentifyMealsList = map(unIdentifyMeal);
+
+const mealMatchesId = curry((id, meal) => meal.id === id);
+
+
+const useMeal = pipe(
+  assoc("lastUsed", Date.now()),
+  when((meal) => !has("usedCount", meal), assoc("usedCount", 0)),
+  evolve({ usedCount: inc })
+);
+
+const findAndUseMeal = function(id, meals) {
+  let useMealWhenIdMatches = when(mealMatchesId(id), useMeal);
+
+  return map(useMealWhenIdMatches, meals);
+};
+
+const updateMeal = curry((changes, meal) => Object.assign(meal, changes));
+
+export {
+  identifyMeal,
+  identifyMealsList,
+  unIdentifyMeal,
+  unIdentifyMealsList,
+  mealMatchesId,
+  useMeal,
+  findAndUseMeal,
+  updateMeal
```

Comment:
many of these are exported just for testing purposes. here are the important ones:

identifyMealsList - add a unique ID property to a list of meals (because I'm not storing them in the storage file with an ID)
unIdentifyMealsList - remove the id prop before saving
findAndUseMeal - find a meal with a given ID in a list of meals and mark it as used
updateMeal - find a meal with a given ID in a list of meals and update it's details, or add a new meal to a list of meals if no ID is specified